pkg/project/admission/lifecycle/admission.go

```diff
@@ -140,10 +141,26 @@ func (e *lifecycle) Handles(operation admission.Operation) bool {
 }
 
 func NewLifecycle(client client.Interface, creatableResources sets.String) (admission.Interface, error) {
```

Comment:
There is no reason why origin admission plugins cannot implement more than just `admission.Interface` and that an additional layer of logic can be done to do more initialization based on those additional interfaces.  The net is that we need the object fully constructed before we start the server, but it does NOT need to be fully instantiated after the base kube call.  For example, if an object needs an origin client, it could implement an `origin.Interface` that says this is how you pass in a client, and we could have an additional code pass that checks for that type of thing.